apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -295,19 +295,19 @@ pick_call(Id, Key, Query, Timeout) ->
     ?PICK(Id, Key, Pid, begin
         Caller = self(),
         MRef = erlang:monitor(process, Pid, [{alias, reply_demonitor}]),
-        From = {Caller, MRef},
+        From = {Caller, [alias | MRef]},
         ReplyTo = {fun gen_statem:reply/2, [From]},
         erlang:send(Pid, ?SEND_REQ(ReplyTo, Query)),
         receive
-            {MRef, Response} ->
+            {[alias | MRef], Response} ->
                 erlang:demonitor(MRef, [flush]),
                 Response;
             {'DOWN', MRef, process, Pid, Reason} ->
                 error({worker_down, Reason})
         after Timeout ->
             erlang:demonitor(MRef, [flush]),
             receive
-                {MRef, Response} ->
+                {[alias | MRef], Response} ->
                     Response
```

Comment:
we are not using gen_statem:call, it's unnecessary to use gen:reply here.
maybe simply 
```
From = Mref,
ReplyTo = {fun erlang:send/2, [Mref]},
erlang:send(Pid, ....)
receive
  {Mref, Response} ->
     ...

Reply:
That's what I did originally.  I'll change it back to that.

Reply:
:heavy_check_mark:

Reply:
Sorry, if it was me suggested to use gen_statem:reply, i probably misunderstood how alias works in the `gen` module.

Reply:
No, I think it was using that since the first version of this module :thinking: